apps/client/src/app/other-resources/other-resources-result/other-resources-result.component.ts

```diff
@@ -0,0 +1,212 @@
+import { ValidatorResponse } from 'fhir/validator';
+
+class OpenedResource {
+  resource: DomainResource;
+  activeSub: 'json/xml'|'permissions' = 'json/xml';
+
+  constructor(resource) {
+    this.resource = resource;
+  }
+}
+
+@Component({
+  templateUrl: './other-resources-result.component.html',
+  styleUrls: ['./other-resources-result.component.css']
+})
+export class OtherResourcesResultComponent implements OnInit {
+  message: string;
+  data: OpenedResource;
+  Globals = Globals;
+  content: string;
+  contentChanged = new EventEmitter();
+  serializationError = false;
+  validation: ValidatorResponse;
+  selected: string = 'JSON';
+  options: string[] = [ 'JSON', 'XML' ];
+
+  constructor(private fhirService: FhirService,
+              private route: ActivatedRoute,
+              private modalService: NgbModal,
+              public configService: ConfigService) {
+
+    this.contentChanged
+      .debounceTime(500)
+      .subscribe(() => {
+        if (this.data) {
+          this.serializationError = false
+          this.message = null;
+          try {
+            // deserialize the content back to data
+            if (this.data.activeSub === 'json/xml' && this.selected === 'JSON') {
+              this.data.resource = JSON.parse(this.content);
+              this.message = 'The content has been updated';
+            } else if (this.data.activeSub === 'json/xml' && this.selected === 'XML') {
+              this.data.resource = this.fhirService.deserialize(this.content);
+              this.message = 'The content has been updated';
+            }
+
+            this.validation = this.fhirService.validate(this.data.resource);
+
+            if(!this.validation.valid){
+              this.message = 'There are validation errors. This resource will still save but please view the validation tab.';
+            }
+
+            setTimeout(() => { this.message = ''; }, 3000);
+          }
+           catch (ex) {
+            this.message = 'The content you have specified is invalid: ' + ex.message;
+            //Deactivate save button when true
+            this.serializationError = true;
+          }
+        }
+      });
+  }
+
+  contentHasChanged(newValue: string) {
+    this.content = newValue;
+    this.contentChanged.emit();
+  }
+
+  ngOnInit() {
+    this.message = 'Opening resource';
+
+    this.fhirService.read(this.route.snapshot.params.type, this.route.snapshot.params.id)
+      .subscribe((results: DomainResource) => {
+
+        this.data = new OpenedResource(results);
+        this.content = JSON.stringify(this.data.resource, null, '\t');
+        this.validation = this.fhirService.validate(this.data.resource);
+
+        setTimeout(() => {
+          this.message = 'Resource opened.';
+        }, 100);
+      }, (err) => {
+        this.message = 'Error opening resource: ' + getErrorString(err);
+      });
+  }
+
+  changeType() {
+    console.log('Changing content type');
```

Comment:
Should a ~500ms delay be put here so that if the user makes changes to JSON, then immediately changes to the type XML, the change has time to reflect in the model? Maybe something like this:
```
setTimeout(() => {
  switch (this.selected) { ... }
}, 500);
```
Maybe that's not beneficial... Just a thought.

Reply:
Yeah I can do that. Would be smart to have.